File: java/src/test/java/org/aa/kata/extractalgo/MessyAlgoTest.java
package org.aa.kata.extractalgo;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class MessyAlgoTest {
    @Test
    void name() {
        assertThat(
                new MessyAlgo().describe(
                        new Article(
                                "chit chat",
                                List.of("ddd", "TDD"),
                                List.of("cyriux", "Jessica Kerr", "aalberti")))
        ).isEqualTo("chit chat is written by Cyrille Martraire (DDD superstar), Jessica Kerr, Antoine Alberti (socio-poetico anarchitect).\n" +
                "It talks about DDD (Domain-Driven Design), TDD");
    }
}

File: kotlin/src/test/kotlin/org/aa/kata/extractalgo/MessyAlgoTest.kt
package org.aa.kata.extractalgo

import org.assertj.core.api.KotlinAssertions.assertThat
import org.junit.jupiter.api.Test

internal class MessyAlgoTest {
    @Test
    internal fun name() {
        assertThat(
            MessyAlgo().describe(
                Article(
                    "chit chat",
                    listOf("ddd", "TDD"),
                    listOf("cyriux", "Jessica Kerr", "aalberti")))
        ).isEqualTo("chit chat is written by Cyrille Martraire (DDD superstar), Jessica Kerr, Antoine Alberti (socio-poetico anarchitect). It talks about DDD (Domain-Driven Design), TDD")
    }
}